sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Blobs/src/AzureBlobXmlRepository.cs

```diff
@@ -71,32 +71,30 @@ public void StoreElement(XElement element, string friendlyName)
             Task.Run(() => StoreElementAsync(element)).GetAwaiter().GetResult();
         }
 
-        private static XDocument CreateDocumentFromBlob(byte[] blob)
+        private static XDocument CreateDocumentFromBlobData(BlobData blobData)
         {
-            using (var memoryStream = new MemoryStream(blob))
+            if (blobData == null || blobData.BlobContents.Length == 0)
```

Comment:
Not obvious in this PR, but can `BlobContents` be null? Just raising the concern in case, but I suspect the answer is "no".

Reply:
Good point. BlobContents is assigned every time BlobData is created.